### PR TITLE
RD-2007 - Improve styling for Label type inputs

### PR DESCRIPTION
**Pull request by qooban**
### Before
image

### After
image

### CODE REVIEW

**Comment by qooban** [comment]
I'm not planning to run full system tests execution. This change affects only `filters_spec`. Executed locally, all tests passed.

**Comment by kubama** [comment]
I have noticed a minor issue:
image

**Comment by kubama** [comment]
Looks like label/attribute and operator dropdowns are now even wider. I think it's a waste of space as there is no option that would require these dropdowns to be that wide. On the other hand the key/value inputs may contain variable length, multiple items. So it would be good if they were wider.

**Comment by qooban** [comment]
@kubama 

> I have noticed a minor issue:
I don't think it is related to this PR. I have seen it before and I'm almost sure it is long with us. I suspect it comes from Semantic-UI-React.
See for example: https://codesandbox.io/embed/semantic-ui-react-example-xghj2 :

https://user-images.githubusercontent.com/5202105/114867185-c2c1e480-9df4-11eb-8666-f0b73eb02d9c.mp4



> Looks like label/attribute and operator dropdowns are now even wider. I think it's a waste of space as there is no option that would require these dropdowns to be that wide. On the other hand the key/value inputs may contain variable length, multiple items. So it would be good if they were wider.

Good point 👍🏻 Makes perfect sense, so I adapted the widths a bit and got something like:

image

What do you think?

File: widgets/common/src/filters/RuleRow.tsx [hunk 1]
```diff
@@ -61,7 +63,7 @@ const RuleRow: FunctionComponent<RuleRowProps> = ({ onChange, onRemove, removabl
                     toolbox={toolbox}
                 />
             </FormField>
-            <FormField>{removable && <RuleRemoveButton onClick={onRemove} />}</FormField>
+            <FormField width={removeButtonFieldWidth}>{removable && <RuleRemoveButton onClick={onRemove} />}</FormField>
```
Comment (Gelio) [inline widgets/common/src/filters/RuleRow.tsx:67]: Optional:

To me, seeing `width={1}` would be as understandable as the current solution, but would provide me with more immediate information and not having to jump to some other place to know the value :smile: 

When it comes to styling, I don't see this as a _magic value_, so IMO hardcoding it here, if it's used once, sounds reasonable :slightly_smiling_face: it's optional, though

Comment (Gelio) [reply widgets/common/src/filters/RuleRow.tsx:67]: The same goes for `7` used as `valueInputFieldWidth` :smile:

**Review by Gelio — DISMISSED** [review]
Looks good from my side :+1:

File: widgets/common/src/filters/RuleRow.tsx (hunk 1 above)
Comment (qooban) [reply widgets/common/src/filters/RuleRow.tsx]: Yep, make sense not to over-complicate it. Originally I had just two consts and then thought it's a bit better, but in that case (3 different consts) it makes it more difficult to read. Changed.

**Comment by kubama** [comment]
> I don't think it is related to this PR. I have seen it before and I'm almost sure it is long with us. I suspect it comes from Semantic-UI-React. See for example: https://codesandbox.io/embed/semantic-ui-react-example-xghj2 :

I have seen it for the first time, but if this is the case then it's of course out of scope of this PR.

> What do you think?

Of course it's better, but it looks like type and operator dropdowns are still wider than needed. Is there any reason for not making them as narrow as possible? In the example screenshot you've given it could make the blueprint value input render without the scroll bar, so I think there is much to gain.

**Comment by qooban** [comment]
> I have seen it for the first time, but if this is the case then it's of course out of scope of this PR.

In this matter I'm 100% sure it is not the first time I see it.
Created: RD-2058 for that and added a comment in https://github.com/Semantic-Org/Semantic-UI-React/issues/3720 .

> Of course it's better, but it looks like type and operator dropdowns are still wider than needed. Is there any reason for not making them as narrow as possible? In the example screenshot you've given it could make the blueprint value input render without the scroll bar, so I think there is much to gain.

I already tried to do so.

With:
```
            <FormField width={3}>
                <RuleRowTypeDropdown onChange={onRuleTypeChange} value={ruleType} />
            </FormField>
            <FormField width={3}>
                <RuleOperatorDropdown onChange={onOperatorChange} value={operator} ruleType={ruleType} />
            </FormField>
```

I get:

image

I suspect that maybe `Dropdown` have a minimal width set somehow, so it would not be so easy to avoid that. If you have a solution for that, please let me know.

**Comment by kubama** [comment]
> I suspect that maybe `Dropdown` have a minimal width set somehow, so it would not be so easy to avoid that. If you have a solution for that, please let me know.

I don't have a solution. In this case I guess we have no option other than to leave it as it is.

**Review by kubama — APPROVED** [review]

**Comment by qooban** [comment]
As this one is small and I applied already @Gelio 's comments, I'll merge it with one approval.
Thanks for the reviews and the discussion around it 👍🏻

**Pull request merged**